LSPS0/README.md

```diff
@@ -0,0 +1,456 @@
+send the message to, and a `msg` parameter, which would be formatted
+similarly to the `payload` of a `custommsg` hook, including the `9419`
+message ID prefix.
+
+### LDK
+
+LDK requires that you create a delegating implementation of the
+`RoutingMessageHandler` trait in order to change the `init` and
+`node_announcement` feature bits for LSPS LSP implementations.
+You should implement a type that contains an instance of the
+LDK-provided `P2PGossipSync` object (or if you want something for
+more general use, itself contains an instance of another object
+with the `RoutingMessageHandler`), and whose implementation
+of `RuotingMessageHandler` simply forwards most of the function
+calls to the contained instance.
+
+The only non-delegating functions would be `provided_node_features`
+and `provided_init_features`.
+The delegating function would call the corresponding function on
+the contained instance, then serialize the returned `Features<T>`
+object via `Writeable::encode`.
+The serialized vector of bytes encodes the 16-bit big-endian length,
+followed by the feature bits vector.
+The feature bits vector needs to be extended to at least 92 bytes,
+packing `0x00` bytes at the *front* of the vector if its length is
+increased, and the byte at `length - 92` should be ORed with `0x2`.
+Then if the length was changed, the 16-bit big-endian length needs
+to be updated.
+The 16-bit big-endian length is concatenated with the feature bits
+vector, then fed into a `Read` object, then a new `InitFeatures`
+or `NodeFeatures` is `Readable::read` from the modified feature
+bits serialization vector.
+
+An LSPS LSP or client must implement a type that implements the
+`CustomMessageHandler` trait, which extends the `CustomMessageReader`
+trait.
+Any `lsps0_message_id` messages would be parsed in the
+`CustomMessageReader::read` function, which is given the message ID
+directly, and the message payload as a `Read`-trait buffer.
+The application would need to define a specialized type to hold
+its own parsed form of custom messages, possibly making the
+`lsps0_message_id` message an `enum` variant.
+
+The specialized type needs to implement `wire::Type`, specifying the
+message ID via the `wire::Type::type_id` function.
+This should check which variant of the custom message type it is and
+return the appropriate message ID.
+The same type is used for both incomiing messages and outgoing
+messages.
+
+Actual handling of incoming requests or responses would be done in the
+`CustomMessageHandler::handle_custom_message`, which is handed the
+custom message type constructed by `CustomMessageReader::read`, as
+well as the node ID of the peer that sent the message.
+
+Outgoing messages will need to be buffered in the type that implements
+`CustomMessageHandler`.
+Periodically, LDK will call `CustomMessageHandler::get_and_clear_pending_msg`,
+which should atomically get the contents of the buffer, empty it, and
+return the contents.
```

Comment:
Some LND methods that are worth mentioning:
- [SendCustomMessage](https://lightning.engineering/api-docs/api/lnd/lightning/send-custom-message/index.html)
- [SubscribeCustomMessages](https://lightning.engineering/api-docs/api/lnd/lightning/subscribe-custom-messages)
- [UpdateNodeAnnouncement](https://lightning.engineering/api-docs/api/lnd/peers/update-node-announcement)

UpdateNodeAnnouncement is not ready yet. See my other comment.

I haven't used the mentioned LND APIs yet so I am hesitant to write up the LND notes. Maybe Breez has some experience there? @yaslama

Reply:
We don't use these APIs either, but they seem to cover all what we need here.

Reply:
testing commenting, please ignore this comment.

Reply:
OK, I will make a very basic section for LND.

Reply:
Looks like there is no way to change the `init` feature bits though?  Seems `UpdateNodeAnnouncement` only changes `node_announcement`?